### PR TITLE
add x123 keymap for crkbd/rev1

### PR DESCRIPTION
**Pull request by x123**
add x123 keymap for crkbd/rev1

This keymap is based on many concepts from Dreymar's big bag theory (see https://dreymar.colemak.org/index.html), tweaked a bit for my own preferences and adopted to the Corne. The readme provides further details and images of the layout.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by dunk2k** [comment]
Somewhat of a contradiction regarding OLED settings here:
defined is OLED font (in `config.h`) and OLED driver (in `rules.mk`) yet `OLED_ENABLE = no`
Could I suggest either removing `OLED_ENABLED` and `OLED_FONT_H` as both are defined higher up and would enable OLED display, *or* removing `OLED_DRIVER` and `OLED_FONT_H` which affirms OLED to be disabled.

**Comment by x123** [comment]
> Somewhat of a contradiction regarding OLED settings here: defined is OLED font (in `config.h`) and OLED driver (in `rules.mk`) yet `OLED_ENABLE = no` Could I suggest either removing `OLED_ENABLED` and `OLED_FONT_H` as both are defined higher up and results in OLED display enabled, _or_ removing `OLED_DRIVER` and `OLED_FONT_H` which affirms OLED display to be disabled.

Thanks for the catch. I've removed the `OLED_DRIVER` and `OLED_FONT_H` lines as you've suggested.

File: keyboards/crkbd/keymaps/x123/keymap.c [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+    case RAISE:
+      if (record->event.pressed) {
+        layer_on(_RAISE);
+        update_tri_layer(_LOWER, _RAISE, _ADJUST);
+      } else {
+        layer_off(_RAISE);
+        update_tri_layer(_LOWER, _RAISE, _ADJUST);
+      }
+      return false;
+      break;
```
Comment (drashna) [inline keyboards/crkbd/keymaps/x123/keymap.c:197]: These can be removed and replaced with the tri layer keycodes
```suggestion
```

https://docs.qmk.fm/#/feature_tri_layer

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/crkbd/keymaps/x123/keymap.c (hunk 1 above)
Comment (x123) [reply keyboards/crkbd/keymaps/x123/keymap.c]: Thanks for the suggestion, I've used it and simplified the keymap configuration significantly. I also removed unused dynamic tapping entries and updated the README to accurately reflect those changes.

**Review by drashna — APPROVED** [review]

File: keyboards/crkbd/keymaps/x123/rules.mk [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
```
Comment (fauxpark) [inline keyboards/crkbd/keymaps/x123/rules.mk:15]:
```suggestion
```

File: keyboards/crkbd/keymaps/x123/keymap.c [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+    case QWERTY:
+      if (record->event.pressed) {
+        set_single_persistent_default_layer(_QWERTY);
+      }
+      return false;
+      break;
```
Comment (fauxpark) [inline keyboards/crkbd/keymaps/x123/keymap.c:148]:
```suggestion
```

File: keyboards/crkbd/keymaps/x123/keymap.c [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+    case QWERTY_NOHOMEROW:
+      if (record->event.pressed) {
+        set_single_persistent_default_layer(_QWERTY_NOHOMEROW);
+      }
+      return false;
+      break;
```
Comment (fauxpark) [inline keyboards/crkbd/keymaps/x123/keymap.c:154]:
```suggestion
```

File: keyboards/crkbd/keymaps/x123/keymap.c [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+    case COLEMAK:
+      if (record->event.pressed) {
+        set_single_persistent_default_layer(_COLEMAK);
+      }
+      return false;
+      break;
```
Comment (fauxpark) [inline keyboards/crkbd/keymaps/x123/keymap.c:160]:
```suggestion
```

File: keyboards/crkbd/keymaps/x123/keymap.c [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+    case COLEMAK_NOHOMEROW:
+      if (record->event.pressed) {
+        set_single_persistent_default_layer(_COLEMAK_NOHOMEROW);
+      }
+      return false;
+      break;
```
Comment (fauxpark) [inline keyboards/crkbd/keymaps/x123/keymap.c:166]:
```suggestion
```

File: keyboards/crkbd/keymaps/x123/rules.mk [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+RGB_MATRIX_ENABLE = no
+RGBLIGHT_ENABLE = no    # Enable WS2812 RGB underlight.
+MOUSEKEY_ENABLE = yes    # Mouse keys
```
Comment (fauxpark) [inline keyboards/crkbd/keymaps/x123/rules.mk:18]:
```suggestion
RGBLIGHT_ENABLE = no
MOUSEKEY_ENABLE = yes
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/crkbd/keymaps/x123/rules.mk (hunk 2 above)
Comment (x123) [reply keyboards/crkbd/keymaps/x123/rules.mk:15]: Additional carriage return added.

File: keyboards/crkbd/keymaps/x123/keymap.c (hunk 4 above)
Comment (x123) [reply keyboards/crkbd/keymaps/x123/keymap.c]: Removed extraneous `break`.

File: keyboards/crkbd/keymaps/x123/keymap.c (hunk 3 above)
Comment (x123) [reply keyboards/crkbd/keymaps/x123/keymap.c]: Removed extraneous `break`.

File: keyboards/crkbd/keymaps/x123/keymap.c (hunk 5 above)
Comment (x123) [reply keyboards/crkbd/keymaps/x123/keymap.c]: Removed extraneous `break`.

File: keyboards/crkbd/keymaps/x123/keymap.c (hunk 6 above)
Comment (x123) [reply keyboards/crkbd/keymaps/x123/keymap.c]: Removed extraneous `break`.

File: keyboards/crkbd/keymaps/x123/rules.mk (hunk 7 above)
Comment (x123) [reply keyboards/crkbd/keymaps/x123/rules.mk]: Removed trailing comments.

File: keyboards/crkbd/keymaps/x123/rules.mk (hunk 2 above)
Comment (fauxpark) [reply keyboards/crkbd/keymaps/x123/rules.mk:15]: That's not what the suggestion is.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by x123** [comment]
I believe I've made all requested changes. Please let me know if anything else is needed.

**Review by fauxpark — APPROVED** [review]

**Comment by zvecr** [comment]
Unfortunately since this PRs creation, QMK has [changed its stance](https://docs.qmk.fm/#/ChangeLog/20230827?id=notable-changes) on accepting user keymaps to the repository.

> qmk/qmk_firmware is no longer accepting PRs for keymaps other than for manufacturer-supported keymaps. User keymap workflow has been documented [here](https://docs.qmk.fm/#/newbs) for several years. This change is to progressively reduce the maintenance burden on the project, and to allow us to focus on the core features of QMK.

In the near future we will be providing a more user focused workflow for keymaps, but till then the suggestion is to keep changes in your own fork.

While we appreciate the time that was taken on this PR, hopefully you understand we are closing it, so that QMK can improve in the long term.

**Pull request closed without merging**